### ISSUE TITLE
Dynamic authorizers should get details

### ISSUE THREAD

**Issue by oberstet** [crossbario/crossbar#430]
Say you want everyone who happens to know or guess the number `0634534534` to be allowed to subscribe to `chat.channel.0634534534`, but disallow everyone from doing a prefix-based subscribe on `chat.channel.`.

In principle, this is a perfect example of why we have dynamic authorizers:

``` python
@wamp.register('com.example.authorize')
def custom_authorize(session, uri, action, options):
   # your custom authorization logic to determine whether client
   # session should be allowed to perform action on uri
   if (action == 'publish' or (action == 'subscribe' and options.get('match', 'exact') == 'exact'))
        and uri.startswith(u'chat.channel.'):
      try:
          int(uri.split('.')[2])
          return True # allow action
      except ValueError:
          return False
   else:
      # deny action
      return False
```

There is only 1 catch: the `options` are currently NOT forwarded to the authorizer (the 4th argument is missing).

**Comment by matthewtorgerson** [comment]
Is there currently any plan to implement this feature?  Not being able to tell if a subscription request is using exact or prefix matching makes the dynamic authorizer a little limited.

**Comment by meejah** [comment]
This has been implemented in #1086
